### PR TITLE
8263644: Add Extract float nodes implementation for Arm SVE

### PR DESCRIPTION
**Pull request by Wanghuang-Huawei**
We implement ExtractD & ExtractF Node for Arm SVE backend.
* overload encoding functions `sve_lasta` & `sve_lastb`
* for future JDK-8263424, we eliminate vector length judgements and use  `rscratch1`

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263644](https://bugs.openjdk.java.net/browse/JDK-8263644): Add Extract float nodes implementation for Arm SVE

### Reviewers
 * [Xiaohong Gong](https://openjdk.java.net/census#xgong) (@XiaohongGong - no project role)
 * [Ningsheng Jian](https://openjdk.java.net/census#njian) (@nsjian - Committer)

### Contributors
 * Wang Huang `<whuang@openjdk.org>`
 * He Xuejin `<hexuejin2@huawei.com>`
 * Ai Jiaming `<aijiaming1@huawei.com>`

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/panama-vector pull/51/head:pull/51`
`$ git checkout pull/51`

To update a local copy of the PR:
`$ git checkout pull/51`
`$ git pull https://git.openjdk.java.net/panama-vector pull/51/head`

### CODE REVIEW

File: src/hotspot/cpu/aarch64/aarch64_sve.ad [hunk 1]
```diff
@@ -3061,3 +3059,37 @@ instruct extractL(iRegLNoSp dst, vReg src, immI idx,  pRegGov pTmp, rFlagsReg cr
...
+
+instruct extractD(vRegD dst, vReg src, immI idx,  pRegGov pTmp, rFlagsReg cr)
+%{
+  predicate(UseSVE > 0);
+  match(Set dst (ExtractD src idx));
+  effect(TEMP pTmp, KILL cr);
+  ins_cost(2 * SVE_COST);
+  format %{ "movzw rscratch1, $idx\n\t"
+            "sve_whilele $pTmp, D, zr, rscratch1\n\t"
+            "sve_lastb $dst, D, $pTmp, $src\n\t" %}
```
Comment (XiaohongGong) [inline src/hotspot/cpu/aarch64/aarch64_sve.ad:3088]: Thanks for the implementation! It mostly looks good to me. Just a trivial comment: add a comment at the end like "extract from vector(D)" like what you have done for` extractB/S` ?

File: src/hotspot/cpu/aarch64/aarch64_sve.ad [hunk 2]
```diff
@@ -3061,3 +3059,37 @@ instruct extractL(iRegLNoSp dst, vReg src, immI idx,  pRegGov pTmp, rFlagsReg cr
   %}
   ins_pipe(pipe_slow);
 %}
+
+instruct extractF(vRegF dst, vReg src, immI idx,  pRegGov pTmp, rFlagsReg cr)
```
Comment (XiaohongGong) [inline src/hotspot/cpu/aarch64/aarch64_sve.ad:3063]: Remove the redundant space between `idx,  pRegGov`?

Comment (Wanghuang-Huawei) [reply src/hotspot/cpu/aarch64/aarch64_sve.ad:3063]: Sure, thank you for your review.

File: src/hotspot/cpu/aarch64/aarch64_sve.ad (hunk 1 above)
Comment (Wanghuang-Huawei) [reply src/hotspot/cpu/aarch64/aarch64_sve.ad:3088]: OK.I will fix that.

**Comment by XiaohongGong** [comment]
LGTM, thanks!

**Review by XiaohongGong — APPROVED** [review]

**Review by nsjian — APPROVED** [review]
Looks good. Thanks!

**Pull request closed without merging**